eZ/Publish/Core/FieldType/RichText/Resources/schemas/docbook/docbook.iso.sch.xsl

```diff
@@ -1,82 +1,311 @@
+        <axsl:apply-templates select="*|comment()|processing-instruction()" mode="M2"/>
+    </axsl:template>
 
-<!--PATTERN -->
+    <!--PATTERN -->
 
 
-	<!--RULE -->
-<axsl:template match="db:link" priority="1000" mode="M3"><svrl:fired-rule xmlns:svrl="http://purl.oclc.org/dsdl/svrl" context="db:link"/>
+    <!--RULE -->
+    <axsl:template match="db:link" priority="1000" mode="M3">
+        <svrl:fired-rule xmlns:svrl="http://purl.oclc.org/dsdl/svrl" context="db:link"/>
 
-		<!--ASSERT -->
-<axsl:choose><axsl:when test="not(contains(@*[name()='xlink:href'], 'javascript:') or contains(@*[name()='xlink:href'], 'vbscript:'))"/><axsl:otherwise><svrl:failed-assert xmlns:svrl="http://purl.oclc.org/dsdl/svrl" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:schold="http://www.ascc.net/xml/schematron" test="not(contains(@*[name()='xlink:href'], 'javascript:') or contains(@*[name()='xlink:href'], 'vbscript:'))"><axsl:attribute name="location"><axsl:apply-templates select="." mode="schematron-get-full-path-2"/></axsl:attribute><svrl:text>using scripts in links is not allowed</svrl:text></svrl:failed-assert></axsl:otherwise></axsl:choose><axsl:apply-templates select="*|comment()|processing-instruction()" mode="M3"/></axsl:template><axsl:template match="text()" priority="-1" mode="M3"/><axsl:template match="@*|node()" priority="-2" mode="M3"><axsl:apply-templates select="*|comment()|processing-instruction()" mode="M3"/></axsl:template></axsl:stylesheet>
+        <!--ASSERT -->
+        <axsl:choose>
+            <axsl:when
+                    test="not(contains(@*[name()='xlink:href'], 'javascript:') or contains(@*[name()='xlink:href'], 'vbscript:'))"/>
+            <axsl:otherwise>
+                <svrl:failed-assert xmlns:svrl="http://purl.oclc.org/dsdl/svrl"
+                                    xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                                    xmlns:schold="http://www.ascc.net/xml/schematron"
+                                    test="not(contains(@*[name()='xlink:href'], 'javascript:') or contains(@*[name()='xlink:href'], 'vbscript:'))">
+                    <axsl:attribute name="location">
+                        <axsl:apply-templates select="." mode="schematron-get-full-path-2"/>
+                    </axsl:attribute>
+                    <svrl:text>using scripts in links is not allowed</svrl:text>
+                </svrl:failed-assert>
+            </axsl:otherwise>
+        </axsl:choose>
+
+        <axsl:choose>
+            <axsl:when test="not(
+                contains(@*[name()='xlink:href'], '&lt;') or
+                contains(@*[name()='xlink:href'], '&gt;') or
+                contains(@*[name()='xlink:href'], '&quot;') or
+                contains(@*[name()='xlink:href'], ' ') or
+                contains(@*[name()='xlink:href'], '&#x09;') or
+                contains(@*[name()='xlink:href'], '&#xA;') or
+                contains(@*[name()='xlink:href'], '&#xD;')
```

Comment:
Note on an implementation itself: I'm not a fan of blacklisting (prefer whitelisting), but since my PR about javascript in hrefs is based also on whitelisting, then I don't object :)
Are you sure this covers every case of possible injection, btw? Not a XSS expert myself ;)